Datasnap/Classes/Utilities/Datasnap.m

```diff
@@ -172,21 +185,20 @@ - (void)onDataInitialized
 - (void)trackEvent:(BaseEvent*)event
```

Comment:
onDataInitialized creates a BaseEvent and sets properties on it and immediately calls trackEvent.   trackEvent then re-sets many of the same properties.  Refactor needed